EXTRA_FIM/main.py

```diff
@@ -175,7 +175,10 @@ def sum_single_k(self, ik):
                     )
 
         # --- write output file
-        filename = f"partial_dos{ik}.h5"
+        if 'path' in kwargs:
+            filename = kwargs['path']+'/'+f"partial_dos{ik}.h5"
+        else:
+            filename = f"partial_dos{ik}.h5"
         with h5py.File(filename, "w") as handle:
             handle.create_dataset(
                 "ionization_energies", data=self.inputDict["ionization_energies"]
```

Comment:
I personally think that sum_single_k should not write a file, but rather return the computed partial result. File writing could be handled outside, e.g. as part of the job that executes the sum_single_k function. This would give a much cleaner code organization.